File: gin-admin-backend/internal/data/role.go
package data

import (
	"time"
)

type Role struct {
	ID        string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	RoleName  string    `gorm:"type:varchar(50);not null" json:"roleName"`
	RoleValue string    `gorm:"type:varchar(50);unique;not null" json:"roleValue"`
	Status    int       `gorm:"type:tinyint;default:1" json:"status"`
	Remark    string    `gorm:"type:varchar(255)" json:"remark"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`

	Menus []Menu `gorm:"many2many:sys_role_menu;joinForeignKey:RoleId;joinReferences:MenuId" json:"menus"`
}

func (Role) TableName() string {
	return "sys_role"
}

File: python-admin-backend/app/models/role.py
from sqlalchemy import Column, String, Integer, Text, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.database import Base
from app.models import sys_role_menu


class Role(Base):
    __tablename__ = "sys_role"

    id = Column(String(36), primary_key=True, index=True)
    role_name = Column(String(50), nullable=False)
    role_value = Column(String(50), unique=True, nullable=False)
    status = Column(Integer, server_default="1")
    remark = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    users = relationship("User", secondary="sys_user_role")
    menus = relationship("Menu", secondary=sys_role_menu)
